### PR TITLE
feat: add BrickwallLimiter, a lookahead limiter with configurable ceiling

### PR DESCRIPTION
**Pull request by cjacoby-sptfy**
`BrickwallLimiter` is a new Limiter plugin for Pedalboard which provides a mastering-like lookahead limiter for preventing peaks above +/-1.0. 

## Problem

Pedalboard's existing `Limiter` plugin wraps JUCE's `dsp::Limiter`, which is a loudness maximizer [1],  but is not so helpful for a more "mastering"-like workflow designed to catch peaks. This `BrickwallLimiter` fills that missing hole.

## Solution

`BrickwallLimiter` plugin with:
- **Configurable ceiling** (`ceiling_db`, default -1.0)
- **Lookahead** with smoothed gain envelope
- **No makeup gain**; signals less than the threshold are unaffected.
- **Optional true-peak detection** via 4× oversampled sidechain (`true_peak=True`).
- **Stereo-linked gain reduction** preserves imaging
- **Streaming-safe**: `prepare()` preserves state for `reset=False` continuations

### Design Notes

* 4-branch gain envelope (attack → hold-refresh → hold → release) with a DelayLine lookahead.
* The hold mechanism ensures gain stays reduced until the triggering peak exits the delay line. Optional 4× Oversampling in the sidechain detects inter-sample peaks without oversampling the audio path.

## Result

```python
from pedalboard import BrickwallLimiter

limiter = BrickwallLimiter(ceiling_db=-1.0, true_peak=True)
output = limiter(audio, sample_rate=44100)
```

[1] This is documented in #282; it compresses, applies automatic makeup gain, and hard-clips at 0 dbFS. 

Disclosure: composed in collaboration with agents, reviewed in detail.

### CODE REVIEW

File: pedalboard_native/__init__.pyi [hunk 1]
```diff
@@ -396,36 +467,6 @@ class Distortion(Plugin):
...
-    ClearsAudioOnReset = 1
-
-    """
-    This plugin type is a bit more of a pain to deal with; it could be argued
-   * that plugins that don't clear their internal buffers when reset() is called
-   * are buggy. To start processing a new buffer, we'll have to find another way
-   * to clear the buffer, usually by reloading the plugin from scratch and
-    persisting its parameters somehow.
-    """
-    PersistsAudioOnReset = 2
```
Comment (caravanuden) [inline pedalboard_native/__init__.pyi:426]: why is this removed?

File: pedalboard_native/__init__.pyi [hunk 2]
```diff
@@ -636,22 +680,13 @@ class ExternalPlugin(Plugin):
...
-    ) -> NDArray[float32]: ...
+    ) -> numpy.ndarray[typing.Any, numpy.dtype[numpy.float32]]: ...
     pass
 
-    @staticmethod
-    def get_plugin_names_for_file(filename: str) -> typing.List[str]:
-        """
-        Return a list of plugin names contained within a given VST3 plugin (i.e.: a ".vst3"). If the provided file cannot be scanned, an ImportError will be raised.
-        """
-        # Not implemented by the abstract base class, but implemented by the concrete subclasses.
```
Comment (caravanuden) [inline pedalboard_native/__init__.pyi:653]: same question as above -- why is this removed?

File: pedalboard_native/__init__.pyi
Comment (caravanuden) [inline pedalboard_native/__init__.pyi:1]: how were the stubs generated? I think there might have been some differences from what is usually done in this repo -- there's a few typing/unrelated changes made throughout the `.pyi` files. found this script for postprocessing after generation with `pybind11-stubgen`  called ` scripts/generate_type_stubs_and_docs.py` -- do we perhaps want to use that?

File: pedalboard/plugins/BrickwallLimiter.h
Comment (caravanuden) [inline pedalboard/plugins/BrickwallLimiter.h:1]: logic here lgtm 👍

File: tests/test_brickwall_limiter.py [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    # analysis/delay bookkeeping is inconsistent, smear/duplicate it.
+    other_samples = np.delete(out_flat, out_peak_idx)
+    assert np.max(np.abs(other_samples)) < 0.05, (
+        "Impulse energy leaked into neighboring samples; upsampling-only "
+        "latency compensation appears miscalibrated."
+    )
+
+
+@pytest.mark.parametrize("n_samples", [1, 3, 10])
+def test_very_short_signal(n_samples: int):
```
Comment (caravanuden) [inline tests/test_brickwall_limiter.py:613]: this only tests short signals below the ceiling -- can we also test a short signal above the ceiling?

File: tests/test_brickwall_limiter.py [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+        limiter.ceiling_db = float("inf")
+
+    # Valid edge cases — must NOT raise
+    limiter.ceiling_db = 0.0  # 0 dBFS ceiling
+    limiter.ceiling_db = 3.0  # positive headroom
+    limiter.ceiling_db = -40.0  # very quiet ceiling
+
+
+@pytest.mark.parametrize("sample_rate", [22050, 44100, 48000])
+@pytest.mark.parametrize("buffer_size", [128, 8192, 65536])
```
Comment (caravanuden) [inline tests/test_brickwall_limiter.py]: for an edge case:
```suggestion
@pytest.mark.parametrize("buffer_size", [1, 128, 8192, 65536])
```

File: tests/test_brickwall_limiter.py [hunk 5]
```diff
@@ -0,0 +1,611 @@
+"""Tests for the BrickwallLimiter plugin."""
```
Comment (caravanuden) [inline tests/test_brickwall_limiter.py:1]: should we add a test for silence?
```python 
def test_silence():
    signal = np.zeros(44100, dtype=np.float32)
    plugin = BrickwallLimiter(ceiling_db=-1.0)
    output = plugin.process(signal, 44100)
    np.testing.assert_array_equal(output, signal)
```

**Review by caravanuden — APPROVED** [review]
nice! thanks for doing this christopher. a few comments/questions; approving so that you're unblocked once my comments are addressed :)

File: tests/test_brickwall_limiter.py (hunk 4 above)
Comment (caravanuden) [reply tests/test_brickwall_limiter.py]: and also do we want to test a couple of really high sample rates; e.g. up to like 200K+?

File: pedalboard_native/__init__.pyi
Comment (cjacoby-sptfy) [reply pedalboard_native/__init__.pyi:1]: blah thanks for cacthing that; i did my self review before noticing i had to regenerate those 😬
